src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java

```diff
@@ -120,7 +120,7 @@ private AuthCredentials extractCredentials0(final RestRequest request) throws El
 
         if (subject == null) {
             log.error("No subject found in JWT token");
-            return null;
+            throw new ElasticsearchSecurityException("No subject found in JWT token", RestStatus.FORBIDDEN);
```

Comment:
Other `HTTPAuthenticator` do not raise `ElasticsearchSecurityException` when they can't validate credentials. Is it possible that authentication header should be processed by another `HTTPAuthenticator`?